ci/date-check/src/main.rs

```diff
@@ -35,50 +36,56 @@ impl fmt::Display for Date {
     }
 }
 
-fn make_date_regex() -> Regex {
-    Regex::new(
-        r"(?x) # insignificant whitespace mode
-        <!--\s*
-        [dD]ate:\s*
-        (?P<y>\d{4}) # year
-        -
-        (?P<m>\d{2}) # month
-        \s*-->",
-    )
-    .unwrap()
+fn make_date_regex() -> Vec<Regex> {
+    let patterns = [
+        r"<!--\s+date-check:\s+(\w+)\s+(\d+{4})\s+-->",
+        r"<!--\s+date-check\s+-->\s+(\w+)\s+(\d+{4})",
+    ];
+    let set = RegexSet::new(&patterns).unwrap();
+    set.patterns()
+        .iter()
+        .map(|pattern| Regex::new(pattern).unwrap())
+        .collect()
 }
 
-fn collect_dates_from_file(date_regex: &Regex, text: &str) -> Vec<(usize, Date)> {
-    let mut line = 1;
-    let mut end_of_last_cap = 0;
-    date_regex
-        .captures_iter(&text)
-        .map(|cap| {
-            (
-                cap.get(0).unwrap().range(),
-                Date {
-                    year: cap["y"].parse().unwrap(),
-                    month: cap["m"].parse().unwrap(),
-                },
-            )
-        })
-        .map(|(byte_range, date)| {
-            line += text[end_of_last_cap..byte_range.end]
-                .chars()
-                .filter(|c| *c == '\n')
-                .count();
-            end_of_last_cap = byte_range.end;
-            (line, date)
-        })
-        .collect()
+fn collect_dates_from_file(date_regexes: &[Regex], text: &str) -> Vec<(usize, Date)> {
+    let mut output = Vec::new();
+    for date_regex in date_regexes {
+        let mut line = 1;
+        let mut end_of_last_cap = 0;
+        let results: Vec<_> = date_regex
+            .captures_iter(text)
+            .filter_map(|cap| {
+                if let (Some(year), Some(month)) = (cap.get(2), cap.get(1)) {
+                    let year = year.as_str().parse().expect("year");
+                    let month = Month::from_str(month.as_str())
+                        .expect("month")
+                        .number_from_month();
+                    Some((cap.get(0).expect("all").range(), Date { year, month }))
+                } else {
+                    None
+                }
+            })
+            .map(|(byte_range, date)| {
+                line += text[end_of_last_cap..byte_range.end]
+                    .chars()
+                    .filter(|c| *c == '\n')
+                    .count();
+                end_of_last_cap = byte_range.end;
+                (line, date)
+            })
+            .collect();
+        output.extend(results);
+    }
+    output
```

Comment:
Please sort the output by line number. Now that two passes are being made the output will be disordered.

Reply:
is this alright